Print REVERTcast as "c :> t" in Printing All mode, and parse it. (#14773)
coq/coq: doc/sphinx/language/core/definitions.rst

.. prodn::
   term_cast ::= @term10 : @type
   | @term10 :> @type
   | @term10 <: @type

>    | @term10 <<: @type

Reviewer: We try to present things in the order of importance to users while also grouping similar items, to the extent possible.  Should this be #2 in the list?  If not, please reorder in `orderedGrammar` and change the order of the descriptions below to match the new order.

The expression :n:`@term10 : @type` is a type cast expression. It enforces
the type of :n:`@term10` to be :n:`@type`.

:n:`@term10 :> @type` has the same effect but may unfold constants in
a different order, resulting in different performance characteristics.
It is primarily intended for use by tactics such as :tacn:`simpl`.

.. flag:: Printing Tactic Casts

>    This flag is off by default. When it is on, the ``:>`` casts are
>    printed, otherwise the term is printed with the cast removed.

Reviewer: ```suggestion
   When set, ``:>`` casts are printed.  Otherwise the term is printed with the cast removed.
   The default is unset.
```
